chain/rust/src/byron/utils.rs

```diff
@@ -0,0 +1,496 @@
+use cbor_event::cbor;
+
+use std::{convert::{TryInto, TryFrom}, fmt};
+use super::*;
+use cml_crypto::{
+    chain_crypto::{self, Sha3_256},
+    Bip32PublicKey, PublicKey,
+    impl_hash_type,
+    CryptoError,
+    RawBytesEncoding,
+};
+use crate::{
+    address::{Address, AddressError},
+    genesis::network_info::NetworkInfo
+};
+use cml_core::{
+    error::{DeserializeError, DeserializeFailure},
+    serialization::{Deserialize, ToBytes}
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum ByronAddressError {
+    #[error("UnknownNetwork: {0}")]
+    UnknownNetwork(ProtocolMagic),
+    #[error("InvalidCRC: found {found}, expected {expected}")]
+    InvalidCRC{ found: Crc32, expected: Crc32 },
+}
+
+impl_hash_type!(AddressId, 28);
+// not sure if this is a hash but it likely is and has the same byte format
```

Comment:
AFAIK it was meant to be a hash and then this feature was never added to mainnet in the end

Reply:
alright I'll just leave it then. whatever it was meant to be, the hash type doesn't have any explicitly "hash" things I guess. it's just a fixed size byte buffer that can be converted to hex/etc.